plugins/chat/assets/javascripts/discourse/components/chat/thread/header.js

```diff
@@ -22,9 +26,44 @@ export default class ChatThreadHeader extends Component {
     );
   }
 
+  get threadNotificationLevel() {
+    return (
+      this.args.thread.currentUserMembership?.notificationLevel ||
+      NotificationLevels.REGULAR
+    );
+  }
+
   @action
   openThreadSettings() {
     const controller = showModal("chat-thread-settings-modal");
     controller.set("thread", this.args.thread);
   }
+
+  @action
+  updateThreadNotificationLevel(val) {
+    const originalVal =
+      this.args.thread.currentUserMembership?.notificationLevel;
+    if (this.args.thread.currentUserMembership) {
+      this.args.thread.currentUserMembership.notificationLevel = val;
```

Comment:
With this I am trying more to do the "optimistic UI updates" we have talked about where we don't wait for the server response to update the UI.